src/java.base/share/classes/javax/crypto/KDFSpi.java

```diff
@@ -77,11 +84,14 @@ protected KDFSpi(KDFParameters kdfParameters)
      *     derivation parameters
      *
      * @return a {@code SecretKey} object corresponding to a key built from the
-     *     KDF output and according to the derivation parameters
+     *     KDF output and according to the derivation parameters or {@code null}
+     *     in cases where an exception is not thrown but a value cannot be
+     *     returned
```

Comment:
It's not clear why an exception is not thrown and return value is null. This also requires users to check for multiple invalid cases.

Reply:
In the era of `Cipher` and `Mac`, we don't treat nullability as a big issue and does not always specify it. Now it's called Java's Billion Dollar Mistake.

In fact, I don't think `doFinal` has ever intended to allow returning null.